common/persistence/sql/sqlExecutionStore.go

```diff
@@ -770,6 +781,12 @@ func (m *sqlExecutionStore) GetTransferTasks(
 			Version:                 info.GetVersion(),
 		}
 	}
+	if len(rows) > 0 {
+		lastTaskID := rows[len(rows)-1].TaskID
+		if lastTaskID < request.MaxReadLevel {
+			resp.NextPageToken = serializePageToken(lastTaskID)
```

Comment:
regarding the comment above, will this adjustment take care of any potential issues?

Reply:
I've updated the tests to cover the new cases.